opensrp-app/src/main/java/org/smartregister/multitenant/ResetAppHelper.java

```diff
@@ -0,0 +1,232 @@
+        preResetAppChecks.add(new SettingsSyncedCheck());
+        preResetAppChecks.add(new StructureSyncedCheck());
+        preResetAppChecks.add(new TaskSyncedCheck());
+    }
+
+    public void startResetProcess(@NonNull AppCompatActivity activity) {
+        resetCancelled = false;
+        // Show some UI here to display the reset progress
+        resetAppDialog = ResetAppDialog.newInstance();
+        resetAppDialog.show(activity.getSupportFragmentManager(), "rest-app-dialog");
+        resetAppDialog.setOnCancelListener((dialogInterface) -> {
+            showProgressText(activity.getString(R.string.cancelling));
+            resetCancelled = true;
+        });
+
+        resetAppDialog.showText(activity.getString(R.string.stopping_services));
+        JobManager.create(application).cancelAll();
+
+        resetAppDialog.showText(activity.getString(R.string.performing_data_checks));
+
+        if (!resetCancelled) {
+            performPreResetChecks();
+        }
+    }
+
+    public void performPreResetChecks() {
+        // Should be handled in the background
+        coreLibraryExecutors.diskIO()
+                .execute(() -> {
+                    try {
+                        for (PreResetAppCheck preResetAppCheck : preResetAppChecks) {
+                            if (!preResetAppCheck.isCheckOk(application)) {
+                                if (resetCancelled) {
+                                    dismissDialog();
+                                    return;
+                                }
+
+                                if (!NetworkUtils.isNetworkAvailable()) {
+                                    dismissDialog();
+                                    coreLibraryExecutors.mainThread().execute(new Runnable() {
+                                        @Override
+                                        public void run() {
+                                            Toast.makeText(application, "No Internet Connection Available", Toast.LENGTH_LONG)
+                                                    .show();
+                                        }
+                                    });
+
+                                    resetCancelled = true;
+                                    return;
+                                }
+
+                                performPreResetOperations(preResetAppCheck);
+                            }
+                        }
+
+                        if (resetCancelled) {
+                            dismissDialog();
+                            return;
+                        }
+
+                        showProgressText(application.getString(R.string.clearing_application_data));
+
+                        if (resetCancelled) {
+                            dismissDialog();
+                            return;
+                        }
+
+                        Timber.e("User %s has completely reset the app", application.getUsername());
```

Comment:
Maybe this should be warning, if the intention is to send to crashlytics

Reply:
I'll change that